File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Stack<Integer> stack = new Stack<>();
        
        // 각 작업의 완료 날짜 계산
        for (int i = 0; i < progresses.length; i++) {
            int remainingWork = 100 - progresses[i];
            int days = (remainingWork + speeds[i] - 1) / speeds[i]; 
            stack.push(days);
        }
        
        List<Integer> answerList = new ArrayList<>();
        while (!stack.isEmpty()) {
            int currentDay = stack.remove(0);
            int count = 1;
            
            while (!stack.isEmpty() && stack.get(0) <= currentDay) {
                count++;
                stack.remove(0);
            }
            
            answerList.add(count);
        }
        
        // 결과 배열로 변환
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    days = []
    
    for progress,speed in zip(progresses,speeds) :
        remain = 100 - progress
        
        if remain % speed == 0 :
            day = remain // speed
        else :
            day = remain // speed + 1
            
        days.append(day)
    
    answer = []
    
    # 앞 기능을 기준으로 함께 배포되는 개수 계산
    
    current_day = days[0]
    count = 1
    
    for day in days[1:] :
        if day <= current_day :
            count += 1
        else :
            answer.append(count)
            current_day = day
            count = 1
            
    answer.append(count)
    return answer
